pkg/controller/clusterconfig/clusterconfig_controller.go

```diff
@@ -108,6 +109,18 @@ func (r *ReconcileClusterConfig) Reconcile(ctx context.Context, request reconcil
 	network.MergeClusterConfig(&operConfig.Spec, clusterConfig.Spec)
 
 	if _, ok := clusterConfig.Annotations[names.NetworkTypeMigrationAnnotation]; ok {
+		// Only allows live migration on SD managed clusters
+		inManagedCluster, err := util.IsRunningInManagedCluster(ctx, r.client)
+		if err != nil {
+			log.Println(err)
+			return reconcile.Result{}, err
+		}
+		if !inManagedCluster {
+			err := fmt.Errorf("Network type live migration is not supported on self-managed clusters")
+			r.status.SetDegraded(statusmanager.ClusterConfig, "NetworkTypeMigrationFailed",
+				fmt.Sprintf("Failed to process network type live migration (%v).Use 'oc edit network.config.openshift.io cluster' to fix.", err))
+			return reconcile.Result{}, err
+		}
 		if hcp := hypershift.NewHyperShiftConfig(); hcp.Enabled {
```

Comment:
~wait what is Hypershift cluster not supported mean? are we differentiating "SD HCP" from "normal HCP"? if so that seems super hacky to me, isn't `hcp.Enabled` set in the managed cluster? again sorry for the questions due to lack of knowledge here.~
NVM, the comment was misleading, this is meant to work only on hosted clusters and not on self-hosted and management clusters...